tests/functional/oauth_sign_up_verification_redirect.js

```diff
@@ -74,28 +79,48 @@ define([
       function countSignInAndInvalidState() {
         return function () {
           return this.parent
-            .findByCssSelector('body')
-              .getVisibleText()
-              .then(function (text) {
-                var isBadRequest = text === 'Bad request - missing code - missing state';
-                if (isBadRequest) {
-                  badRequestCount++;
-                }
-              })
-            .end()
-
-            .setFindTimeout(0)
-            .findByCssSelector('#loggedin')
-              .then(function () {
-                signedInCount++;
-              }, function (err) {
-                if (/NoSuchElement/.test(String(err))) {
-                  // swallow the error
-                  return;
-                }
-
-                throw err;
-              })
+            // This test can do redirecting. We want to make sure all redirections
+            // have completed. The user will either finish at 123done's logged in
+            // screen, or 123done's /api/oauth endpoint where an error message will
+            // be displayed.
+            // As ugly as it is, wait a few seconds for the redirects to finish.
+            // Check which URL the user is at. If they are at the /api/oauth endpoint,
+            // check for the error message. If they are at 123done, then check for
+            // the #loggedin element.
+
+            .sleep(5000) // <--- GROSS!!! Read above.
```

Comment:
I guess we have something like that here https://github.com/mozilla/fxa-content-server/pull/4869/files#diff-15111e5ba80b38e731a31994bd2fba16R1651 but in this case it could be a custom poll